File: api/src/main/java/com/mystore/manager/api/business/common/criteria/CaisseSessionCriteria.java
package com.mystore.manager.api.business.common.criteria;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.mystore.manager.api.common.criteria.PaginationCriteria;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@JsonIgnoreProperties(ignoreUnknown = true)
public class CaisseSessionCriteria extends PaginationCriteria {
    private Integer posId;
    private String status;
    private String fromDate;
    private String toDate;
}

File: ui-admin/src/app/backend/criteria/business/caisse-session-criteria.ts
import { CommonCriteria } from '../commoncriteria';

export class CaisseSessionCriteria extends CommonCriteria {
    posId: number;
    status: string;
    fromDate: string;
    toDate: string;

    constructor() {
        super();
        this.posId = null;
        this.status = null;
        this.fromDate = null;
        this.toDate = null;
    }
}
